From 9a49cbdff3bc2a1505032aed5ca8129163e40ea5 Mon Sep 17 00:00:00 2001
From: Zi Shen Lim <zlim@netlogicmicro.com>
Date: Thu, 29 Sep 2011 15:46:51 -0700
Subject: [PATCH 373/762] Resolve merge conflict: pick linux-3.0-stable

why was this commented out? revisit.

Based on Broadcom SDK 2.3.

Signed-off-by: Zi Shen Lim <zlim@netlogicmicro.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 crypto/testmgr.h |   66 +++++++++++++++++++++++++++---------------------------
 1 files changed, 33 insertions(+), 33 deletions(-)

diff --git a/crypto/testmgr.h b/crypto/testmgr.h
index 3d26b9b..36e5a8e 100644
--- a/crypto/testmgr.h
+++ b/crypto/testmgr.h
@@ -142,13 +142,13 @@ static struct hash_testvec md4_tv_template [] = {
 /*
  * MD5 test vectors from RFC1321
  */
-#define MD5_TEST_VECTORS	5
+#define MD5_TEST_VECTORS	7
 
 static struct hash_testvec md5_tv_template[] = {
-/*	{
+	{
 		.digest	= "\xd4\x1d\x8c\xd9\x8f\x00\xb2\x04"
 			  "\xe9\x80\x09\x98\xec\xf8\x42\x7e",
-	},*/ {
+	}, {
 		.plaintext = "a",
 		.psize	= 1,
 		.digest	= "\x0c\xc1\x75\xb9\xc0\xf1\xb6\xa8"
@@ -175,13 +175,13 @@ static struct hash_testvec md5_tv_template[] = {
 		.psize	= 62,
 		.digest	= "\xd1\x74\xab\x98\xd2\x77\xd9\xf5"
 			  "\xa5\x61\x1c\x2c\x9f\x41\x9d\x9f",
-	}, /*{
+	}, {
 		.plaintext = "12345678901234567890123456789012345678901234567890123456789012"
 			   "345678901234567890",
 		.psize	= 80,
 		.digest	= "\x57\xed\xf4\xa2\x2b\xe3\xc9\x55"
 			  "\xac\x49\xda\x2e\x21\x07\xb6\x7a",
-	}*/
+	}
 
 };
 
@@ -1059,7 +1059,7 @@ static struct hash_testvec ghash_tv_template[] =
  * HMAC-MD5 test vectors from RFC2202
  * (These need to be fixed to not use strlen).
  */
-#define HMAC_MD5_TEST_VECTORS	5
+#define HMAC_MD5_TEST_VECTORS	7
 
 static struct hash_testvec hmac_md5_tv_template[] =
 {
@@ -1108,7 +1108,7 @@ static struct hash_testvec hmac_md5_tv_template[] =
 		.psize	= 20,
 		.digest	= "\x56\x46\x1e\xf2\x34\x2e\xdc\x00"
 			  "\xf9\xba\xb9\x95\x69\x0e\xfd\x4c",
-	}, /*{
+	}, {
 		.key	= "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
@@ -1135,7 +1135,7 @@ static struct hash_testvec hmac_md5_tv_template[] =
 		.psize	= 73,
 		.digest	= "\x6f\x63\x0f\xad\x67\xcd\xa0\xee"
 			  "\x1f\xb1\xf5\x62\xdb\x3a\xa5\x3e",
-	},*/
+	},
 };
 
 /*
@@ -1303,7 +1303,7 @@ static struct hash_testvec hmac_rmd160_tv_template[] = {
 /*
  * HMAC-SHA1 test vectors from RFC2202
  */
-#define HMAC_SHA1_TEST_VECTORS	5
+#define HMAC_SHA1_TEST_VECTORS	7
 
 static struct hash_testvec hmac_sha1_tv_template[] = {
 	{
@@ -1352,7 +1352,7 @@ static struct hash_testvec hmac_sha1_tv_template[] = {
 		.psize	= 20,
 		.digest	= "\x4c\x1a\x03\x42\x4b\x55\xe0\x7f\xe7\xf2"
 			  "\x7b\xe1\xd5\x8b\xb9\x32\x4a\x9a\x5a\x04",
-	}, /*{
+	}, {
 		.key	= "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
@@ -1379,7 +1379,7 @@ static struct hash_testvec hmac_sha1_tv_template[] = {
 		.psize	= 73,
 		.digest	= "\xe8\xe9\x9d\x0f\x45\x23\x7d\x78\x6d\x6b"
 			  "\xba\xa7\x96\x5c\x78\x08\xbb\xff\x1a\x91",
-	},*/
+	},
 };
 
 
@@ -1501,7 +1501,7 @@ static struct hash_testvec hmac_sha224_tv_template[] = {
  * HMAC-SHA256 test vectors from
  * draft-ietf-ipsec-ciph-sha-256-01.txt
  */
-#define HMAC_SHA256_TEST_VECTORS	8
+#define HMAC_SHA256_TEST_VECTORS	10
 
 static struct hash_testvec hmac_sha256_tv_template[] = {
 	{
@@ -1604,7 +1604,7 @@ static struct hash_testvec hmac_sha256_tv_template[] = {
 			  "\x1a\xb9\xc3\x74\x9a\x5f\x1c\x17"
 			  "\xd4\xf5\x89\x66\x8a\x58\x7b\x27"
 			  "\x00\xa9\xc9\x7c\x11\x93\xcf\x42",
-	}, /*{
+	}, {
 		.key	= "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
@@ -1635,13 +1635,13 @@ static struct hash_testvec hmac_sha256_tv_template[] = {
 			  "\xc8\x48\x1a\x5c\xa4\x82\x5b\xc8"
 			  "\x84\xd3\xe7\xa1\xff\x98\xa2\xfc"
 			  "\x2a\xc7\xd8\xe0\x64\xc3\xb2\xe6",
-	},*/
+	},
 };
 
-#define XCBC_AES_TEST_VECTORS 5
+#define XCBC_AES_TEST_VECTORS 6
 
 static struct hash_testvec aes_xcbc128_tv_template[] = {
-	/*{
+	{
 		.key	= "\x00\x01\x02\x03\x04\x05\x06\x07"
 			  "\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f",
 		.plaintext = zeroed_string,
@@ -1649,7 +1649,7 @@ static struct hash_testvec aes_xcbc128_tv_template[] = {
 			  "\x45\x73\xdf\xd5\x84\xd7\x9f\x29",
 		.psize	= 0,
 		.ksize	= 16,
-	},*/ {
+	}, {
 		.key	= "\x00\x01\x02\x03\x04\x05\x06\x07"
 			  "\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f",
 		.plaintext = "\x00\x01\x02",
@@ -1782,7 +1782,7 @@ static struct hash_testvec aes_vmac128_tv_template[] = {
  * SHA384 HMAC test vectors from RFC4231
  */
 
-#define HMAC_SHA384_TEST_VECTORS	2
+#define HMAC_SHA384_TEST_VECTORS	4
 
 static struct hash_testvec hmac_sha384_tv_template[] = {
 	{
@@ -1811,7 +1811,7 @@ static struct hash_testvec hmac_sha384_tv_template[] = {
 			  "\x8b\x32\x39\xec\xfa\xb2\x16\x49",
 		.np	= 4,
 		.tap	= { 7, 7, 7, 7 }
-	},/* {
+	}, {
 		.key	= "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			  "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			  "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
@@ -1875,14 +1875,14 @@ static struct hash_testvec hmac_sha384_tv_template[] = {
 			  "\xdc\xce\xbb\x82\x46\x1e\x99\xc5"
 			  "\xa6\x78\xcc\x31\xe7\x99\x17\x6d"
 			  "\x38\x60\xe6\x11\x0c\x46\x52\x3e",
-	},*/
+	},
 };
 
 /*
  * SHA512 HMAC test vectors from RFC4231
  */
 
-#define HMAC_SHA512_TEST_VECTORS	2
+#define HMAC_SHA512_TEST_VECTORS	4
 
 static struct hash_testvec hmac_sha512_tv_template[] = {
 	{
@@ -1915,7 +1915,7 @@ static struct hash_testvec hmac_sha512_tv_template[] = {
 			  "\x63\x6e\x07\x0a\x38\xbc\xe7\x37",
 		.np	= 4,
 		.tap	= { 7, 7, 7, 7 }
-	},/* {
+	}, {
 		.key	= "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			  "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			  "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
@@ -1986,7 +1986,7 @@ static struct hash_testvec hmac_sha512_tv_template[] = {
 			"\x0d\x5e\xeb\x55\xc3\xe4\xde\x15"
 			"\x13\x46\x76\xfb\x6d\xe0\x44\x60"
 			"\x65\xc9\x74\x40\xfa\x8c\x6a\x58",
-	},*/
+	},
 };
 
 /*
@@ -6119,16 +6119,16 @@ static struct cipher_testvec cast6_dec_tv_template[] = {
 #define AES_CTR_DEC_TEST_VECTORS 3
 #define AES_OFB_ENC_TEST_VECTORS 1
 #define AES_OFB_DEC_TEST_VECTORS 1
-#define AES_CTR_3686_ENC_TEST_VECTORS 4
-#define AES_CTR_3686_DEC_TEST_VECTORS 4
+#define AES_CTR_3686_ENC_TEST_VECTORS 7
+#define AES_CTR_3686_DEC_TEST_VECTORS 6
 #define AES_GCM_ENC_TEST_VECTORS 9
 #define AES_GCM_DEC_TEST_VECTORS 8
 #define AES_GCM_4106_ENC_TEST_VECTORS 7
 #define AES_GCM_4106_DEC_TEST_VECTORS 7
 #define AES_CCM_ENC_TEST_VECTORS 7
 #define AES_CCM_DEC_TEST_VECTORS 7
-#define AES_CCM_4309_ENC_TEST_VECTORS 1
-#define AES_CCM_4309_DEC_TEST_VECTORS 1
+#define AES_CCM_4309_ENC_TEST_VECTORS 7
+#define AES_CCM_4309_DEC_TEST_VECTORS 10
 
 static struct cipher_testvec aes_enc_tv_template[] = {
 	{ /* From FIPS-197 */
@@ -7768,7 +7768,7 @@ static struct cipher_testvec aes_ctr_rfc3686_enc_tv_template[] = {
 			  "\x84\x90\x70\x1c\x5a\xd4\xa7\x9c"
 			  "\xfc\x1f\xe0\xff\x42\xf4\xfb\x00",
 		.rlen	= 32,
-	}, /*{
+	}, {
 		.key	= "\x77\x6b\xef\xf2\x85\x1d\xb0\x6f"
 			  "\x4c\x8a\x05\x42\xc8\x69\x6f\x6c"
 			  "\x6a\x81\xaf\x1e\xec\x96\xb4\xd3"
@@ -8840,7 +8840,7 @@ static struct cipher_testvec aes_ctr_rfc3686_enc_tv_template[] = {
 		.rlen = 4100,
 		.np	= 2,
 		.tap	= { 4064, 36 },
-	},*/
+	},
 };
 
 static struct cipher_testvec aes_ctr_rfc3686_dec_tv_template[] = {
@@ -8900,7 +8900,7 @@ static struct cipher_testvec aes_ctr_rfc3686_dec_tv_template[] = {
 			  "\x10\x11\x12\x13\x14\x15\x16\x17"
 			  "\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f",
 		.rlen	= 32,
-	}, /*{
+	}, {
 		.key	= "\x77\x6b\xef\xf2\x85\x1d\xb0\x6f"
 			  "\x4c\x8a\x05\x42\xc8\x69\x6f\x6c"
 			  "\x6a\x81\xaf\x1e\xec\x96\xb4\xd3"
@@ -8931,7 +8931,7 @@ static struct cipher_testvec aes_ctr_rfc3686_dec_tv_template[] = {
 			  "\x10\x11\x12\x13\x14\x15\x16\x17"
 			  "\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f",
 		.rlen	= 32,
-	},*/
+	},
 };
 
 static struct cipher_testvec aes_ofb_enc_tv_template[] = {
@@ -10133,7 +10133,7 @@ static struct aead_testvec aes_ccm_rfc4309_enc_tv_template[] = {
 };
 
 static struct aead_testvec aes_ccm_rfc4309_dec_tv_template[] = {
-	/*{
+	{
 		.key	= "\xab\x2f\x8a\x74\xb7\x1c\xd2\xb1"
 			  "\xff\x80\x2e\x48\x7d\x82\xf8\xb9"
 			  "\xc6\xfb\x7d",
@@ -10156,7 +10156,7 @@ static struct aead_testvec aes_ccm_rfc4309_dec_tv_template[] = {
 		.ilen	= 8,
 		.result	= "\x00",
 		.rlen	= 0,
-	},*/ {
+	}, {
 		.key	= "\x61\x0e\x8c\xae\xe3\x23\xb6\x38"
 			  "\x76\x1c\xf6\x3a\x67\xa3\x9c\xd8"
 			  "\xc6\xfb\x7d",
-- 
1.7.0.4

